c++/src/kj/common.h

```diff
@@ -1928,6 +1929,51 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return true;
   }
 
+  inline auto operator<=>(decltype(nullptr)) const { 
+    using ReturnType = decltype(ptr[0] <=> ptr[0]);
+    return size_ == 0 ? (ReturnType) std::strong_ordering::equal
+                      : (ReturnType) std::strong_ordering::greater;
+  }
+
+  inline auto operator<=>(const ArrayPtr& other) const {
+    //Lexographic 3-way comparitor function 
```

Comment:
comparitor => comparator, but I think official name is "three-way comparison operator"

Reply:
It's important to specify it is lexographic, otherwise it is ambiguous if the size is checked first.